### DISCUSSION TITLE
Metadata is getting lost

### DISCUSSION THREAD

**Discussion by apatenge-bhs** [open-feature/dotnet-sdk#280] [Q&A] [answered]
Hi OpenFeature .NET team, 

I am wondering if there is a bug in the handling of the meta data between the resolver and the feature client interface.

I have created an own provider for my system and when I handover the metadata to the **_ResolutionDetails_** in the resolver it disappears on the way to the feature client.

I guess I found the reason for it. In the **_ResolutionDetailsExtensions_** the meta data is not passed through. It seems it is forgotten be added to the parameters of the **_FlagEvaluationDetails_** there.

See here: [https://github.com/open-feature/dotnet-sdk/blob/main/src/OpenFeature/Extension/ResolutionDetailsExtensions.cs](https://github.com/open-feature/dotnet-sdk/blob/main/src/OpenFeature/Extension/ResolutionDetailsExtensions.cs)

I got me the source code of the latest release 1.5.1 and changed it as the following:

From:
`return new FlagEvaluationDetails<T>(details.FlagKey, details.Value, details.ErrorType, details.Reason,
                details.Variant, details.ErrorMessage);`

To:
`return new FlagEvaluationDetails<T>(details.FlagKey, details.Value, details.ErrorType, details.Reason,
                details.Variant, details.ErrorMessage, details.FlagMetadata);`

With that change it seems to work properly.

Could you please check for this and prove my change? Maybe I am wrong and my way to use it is wrong.

If not and this is correct, would there be a possibility to release a new version (1.5.2 ?!) with that change and provide this as Nuget package?

I would like to use the official package instead of my custom version.

Many thanks.

Regards,
André

**Comment by askpt** [comment, marked as answer]
Hello,
Thank you for reporting this. I will create an issue, and we should release a version soon to fix this. You can follow the developments here: https://github.com/open-feature/dotnet-sdk/issues/281

**Reply by beeme1mr** [reply]
This has been addressed in [1.5.2](https://github.com/open-feature/dotnet-sdk/releases/tag/v1.5.2).

**Comment by apatenge-bhs** [comment]
Many thanks for this. I will test it with my application now.

**Comment by apatenge-bhs** [comment]
This change was working as expected. Many thanks.
